api/rosetta/data_block_integration_test.go

```diff
@@ -0,0 +1,563 @@
+		},
+		{
+			// initial transfer of currency from the root account to the user - 100 tokens
+			name:                 "block mid-chain with transactions",
+			request:              blockRequest(13, knownBlockID(13)),
+			wantTimestamp:        1621338403243,
+			wantParentHash:       knownBlockID(12),
+			transactionValidator: validateSingleTransfer(t, "a9c9ab28ea76b7dbfd1f2666f74348e4188d67cf68248df6634cee3f06adf7b1", "8c5303eaa26202d6", "754aed9de6197641", 100_00000000),
+		},
+		{
+			name:           "block mid-chain without transactions",
+			request:        blockRequest(43, knownBlockID(43)),
+			wantTimestamp:  1621341103243,
+			wantParentHash: knownBlockID(42),
+		},
+		{
+			// transaction between two users
+			name:                 "second block mid-chain with transactions",
+			request:              blockRequest(44, knownBlockID(44)),
+			wantTimestamp:        1621341193243,
+			wantParentHash:       knownBlockID(43),
+			transactionValidator: validateSingleTransfer(t, "d5c18baf6c8d11f0693e71dbb951c4856d4f25a456f4d5285a75fd73af39161c", "754aed9de6197641", "631e88ae7f1d7c20", 1),
+		},
+		{
+			name: "lookup of a block mid-chain by index only",
+			request: rosetta.BlockRequest{
+				NetworkID: defaultNetworkID(),
+				BlockID:   identifier.Block{Index: 44}},
+			wantTimestamp:          1621341193243,
+			wantParentHash:         knownBlockID(43),
+			transactionValidator:   validateSingleTransfer(t, "d5c18baf6c8d11f0693e71dbb951c4856d4f25a456f4d5285a75fd73af39161c", "754aed9de6197641", "631e88ae7f1d7c20", 1),
+			customBlockIDValidator: validateBlockID(t, 44, knownBlockID(44)), // verify that the returned block ID has both height and hash
+		},
+		{
+			name:           "last indexed block",
+			request:        blockRequest(425, knownBlockID(425)),
+			wantTimestamp:  1621375483243,
+			wantParentHash: knownBlockID(424),
+		},
+	}
+
+	for _, test := range tests {
+
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+
+			t.Parallel()
+
+			enc, err := json.Marshal(test.request)
+			require.NoError(t, err)
+
+			req := httptest.NewRequest(http.MethodPost, "/block", bytes.NewReader(enc))
+			req.Header.Set(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+			rec := httptest.NewRecorder()
+
+			ctx := echo.New().NewContext(req, rec)
+
+			err = api.Block(ctx)
+			assert.NoError(t, err)
+
+			// unpack response
+			var blockResponse rosetta.BlockResponse
+			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &blockResponse))
+
+			if assert.NotNil(t, blockResponse.Block) {
+
+				// verify block ID of the returned block either using the provided validator,
+				// or by comparing the requested and returned block height and hash
+				if test.customBlockIDValidator != nil {
+					test.customBlockIDValidator(blockResponse.Block.ID)
+				} else {
+					assert.Equal(t, test.request.BlockID.Index, blockResponse.Block.ID.Index)
+					assert.Equal(t, test.request.BlockID.Hash, blockResponse.Block.ID.Hash)
+				}
```

Comment:
You could turn this into a single line by declaring a default block validator that checks `Index` and `Hash` against a reference, and set it for those cases when declaring them.